### PR TITLE
Rating

### PR DESCRIPTION
**Pull request by Allislove**

### CODE REVIEW

File: src/pages/exam.tsx [hunk 1]
```diff
@@ -102,7 +106,21 @@ const Exam = () => {
...
+    let points = 0
+    let average = 0
+    questions.forEach((question: Question) => {
+      const selectedOption = selectedOptions[question.question as keyof typeof selectedOptions]
+      const correctOption = question.options.find((option) => option.points > 0)
+      if (selectedOption === correctOption?.option) {
+        points += correctOption.points
+        average = points / questions.length
+      }
+    })
```
Comment (Israel-Laguan) [inline src/pages/exam.tsx:114]: What about we make this sum when the user press "Next" button?

Comment (Allislove) [reply src/pages/exam.tsx:114]: Yeah, i was thinking about it too, but, what about the modal?? Remember that? we need to show the modal! or what about that ?

File: src/pages/exam.tsx [hunk 2]
```diff
@@ -122,67 +140,70 @@ const Exam = () => {
...
+                  Next
+                </button>
+              )}
+              {currentQuestionIndex === questions.length - 1 && (
+                <button onClick={surveyEnd} type="button">
+                  Finish
+                </button>
+              )}
+            </div>
+          </form>
```
Comment (Israel-Laguan) [inline src/pages/exam.tsx]: Separate form in another component

Comment (Allislove) [reply src/pages/exam.tsx]: Ok

File: src/pages/exam.tsx (hunk 1 above)
Comment (Israel-Laguan) [reply src/pages/exam.tsx:114]: And the final calculation logic can be on Result component

Comment (Allislove) [reply src/pages/exam.tsx:114]: Oh ok

Comment (Israel-Laguan) [reply src/pages/exam.tsx:114]: In general, you should use the context too

Comment (Israel-Laguan) [reply src/pages/exam.tsx:114]: What if the modal is call from result component?

Comment (Allislove) [reply src/pages/exam.tsx:114]: I need to test this.

**Comment by Israel-Laguan** [comment]
Mira no puede buildear la pagina form
image

File: src/pages/exam.tsx (hunk 2 above)
Comment (Allislove) [reply src/pages/exam.tsx]: Done,

File: src/pages/exam.tsx (hunk 1 above)
Comment (Allislove) [reply src/pages/exam.tsx:114]: Done

File: src/pages/exam.tsx [hunk 3]
```diff
@@ -91,18 +97,38 @@ const Exam = () => {
...
     if (!isOptionSelected) {
       notifyError('¡Selecciona la respuesta!')
       return
     }
-    window.location.href = '/exam' // Pending
+
+    console.log('Clicked')
+    surveyResults()
+    setShowModal(true)
+    //window.location.href = '/results' // Pending
```
Comment (Israel-Laguan) [inline src/pages/exam.tsx]: remove this if not needed

**Pull request merged**